packages/leva/src/components/UI/Joystick.tsx

```diff
@@ -1,16 +1,21 @@
 import React, { useState, useRef, useCallback, useEffect, useLayoutEffect } from 'react'
 import { useDrag } from '../../hooks'
 import { clamp, multiplyStep } from '../../utils'
-import { JoystickTrigger, JoystickPlayground } from './StyledJoystick'
+import { JoystickTrigger, JoystickPlayground, JoystickGrid } from './StyledJoystick'
+import { JoystickPlayground3d } from './JoystickPlayground3d'
 import { useTh } from '../../styles'
-import { Portal } from '../UI'
+import { Portal } from '.'
 import { useTransform } from '../../hooks'
-import type { Vector2d } from '../../types'
-import type { Vector2dProps } from './vector2d-types'
+import type { Vector2d, Vector3d } from '../../types'
+import type { Vector2dProps } from '../Vector2d/vector2d-types'
+import { useKeyPress } from '../../hooks/useKeyPress'
 
-type JoystickProps = { value: Vector2d } & Pick<Vector2dProps, 'onUpdate' | 'settings'>
+type JoystickProps = { value: Vector2d | Vector3d } & Pick<Vector2dProps, 'onUpdate' | 'settings'> & { children?: any }
+
+export function Joystick({ value, settings, onUpdate, children }: JoystickProps) {
+  const ctrlKey = useKeyPress('Control')
+  const metaKey = useKeyPress('Meta')
```

Comment:
This smells... I'm redeclaring control keys for the joystick and then passing them down into JoystickPlayground3d which passes them down into the JoystickCube. I should only declare them once

Reply:
Removed these